referenceframe/frame.go

```diff
@@ -121,6 +121,9 @@ type Frame interface {
 	// ProtobufFromInput does there correct thing for this frame to convert input units (radians/mm) to protobuf units (degrees/mm)
 	ProtobufFromInput([]Input) *pb.JointPositions
 
+	// InputFromMap looks through a map and returns a slice of Inputs corresponding to the given frame's name
+	InputFromMap(map[string][]Input) ([]Input, error)
```

Comment:
We've already got the public function `GetFrameInputs` that does exactly this.

We shouldn't be putting methods on interfaces to do super trivial convenience functions like this. Every time we do, it makes it harder to implement anything that doesn't inherit from the private base struct.

Reply:
Currently we have to use switch statements to disambiguate the solver frame from other frames.  Its way uglier to have to call out the case (in a few places of the codebase at this point) where a frame might not be supported by GetFrameInputs, than the small overhead we incur if we happen to write a frame that doesn't inherit from `baseFrame`

Reply:
This change rectifies things so that you can pass around the more general `Frame`, and not have to worry about what type of frame it is

Reply:
Thinking about this more, there could be a middle ground in which we keep GetFrameInputs and do the case checking there

Reply:
Looked into the above option and it's also a little weird.  We would like to be able to call this from `referenceframe`, but since `solverFrames` exist outside this package we can't do the type check.  I'm back to thinking that adding the method is the cleanest solution

Reply:
The only reason this change is needed, is that the planager was updated to take a generic frame rather than a solver frame.

Since we have tightly controlled entry into all planning now via the planager, there's no need for this. All input frames are wrapped in solver frames.

We should undo the generic frame change, swap all planning back to operating on solverFrames (wrapping frames with solver frames if need be), and then we don't need this at all.

Reply:
I started writing a long thing about solverFrames and the extent of their lifetimes, yadda, yadda.

It doesn't matter. Peter is right. It's far more sustainable to constrain `solverFrame` and all its structural nuance to the `motionplan` package and do sanitization of input frames so the user doesn't have to start keeping multiple frame styles within their mind.